seir-forecast/seir-forecast.ipynb

```diff
@@ -0,0 +1,394 @@
+    "        yCim = [sum(i) for i in zip(yEim, yIim)]  \n",
+    "        res = sum(\n",
+    "             1 * np.subtract(yCim, self.cases) ** 2\n",
+    "            + np.subtract(yRim, self.deaths) ** 2\n",
+    "        )\n",
+    "        return res\n",
+    "\n",
+    "    def optimize(self, params):\n",
+    "        \"\"\"Describe what's happening here.\n",
+    "        \n",
+    "        Args:\n",
+    "            params: Define.\n",
+    "        \n",
+    "        Returns:\n",
+    "            Describe what is returned.\n",
+    "        \"\"\"\n",
+    "        res = optimize.minimize(\n",
+    "            self.error,\n",
+    "            params,\n",
+    "            method = \"L-BFGS-B\",\n",
+    "            bounds = [(0.0, 20.0), (0.0, 20.0), (0.0, 20.0)],\n",
+    "            options = {\"xtol\": 1e-8, \"disp\": True, \"ftol\": 1e-8, \"maxiter\": 1e7},\n",
+    "        )\n",
+    "        return res"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run optimization"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set up population distribution in initial state\n",
+    "N = 5180493\n",
+    "S_0 = (N - 11) / N\n",
+    "E_0 = 10 / N\n",
+    "I_0 = 1 / N\n",
+    "R_0 = 0\n",
+    "init_vals = [S_0, E_0, I_0, R_0]\n",
+    "\n",
+    "# Parameter reported by researchers\n",
+    "epsilon, beta, gamma = [0.2, 1.75, 0.5]\n",
+    "params = epsilon, beta, gamma\n",
+    "\n",
+    "# Instantiate the class\n",
+    "seir_eval = OptimizeParameters(init_vals, df.cases / N, df.deaths / N)\n",
+    "\n",
+    "# Run optimiza function\n",
+    "opt_p = seir_eval.optimize(params)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Run simulation using optimized parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# define time interval\n",
+    "t_max = 200\n",
+    "dt = 1\n",
+    "t = np.linspace(0, t_max, int(t_max / dt) + 1)\n",
+    "days = len(df.cases) + 14\n",
+    "\n",
+    "results = base_seir_model(init_vals, opt_p.x, t)\n",
+    "\n",
+    "simulated_cases = results[:days,1] + results[:days,2]\n",
+    "observed_cases = df.cases/N\n",
+    "\n",
+    "times = [date2num(s) for (s) in df.date]\n",
+    "obs_dates = list(map(lambda x: num2date(x) , times)) \n",
+    "extra_times=np.linspace(times[-1]+1.0,times[-1] + 14.0,14)\n",
+    "times.extend(extra_times)\n",
+    "sim_dates = list(map(lambda x: num2date(x) , times)) \n",
+    "\n",
+    "obs = pd.DataFrame({\"observed_cases\":observed_cases,\"date\":obs_dates})\n",
+    "sim = pd.DataFrame({\"simulated_cases\":simulated_cases,\"date\":sim_dates})\n",
+    "\n",
+    "obs.to_csv(\"simulated_cases.txt\",index=False,sep = '\\t')\n",
+    "sim.to_csv(\"observed_cases.txt\",index=False,sep = '\\t')"
```

Comment:
i already told Yilin about this so she will probably push an update - but "simulated_cases.txt" contains the observed cases and vice versa

Reply:
Fixed. I probably missed the discussion yesterday.